packages/js/product-editor/src/blocks/variations/edit.tsx

```diff
@@ -101,15 +103,22 @@ export function Edit( {
 	};
 
 	const handleAdd = ( newOptions: EnhancedProductAttribute[] ) => {
-		handleChange( [
-			...newOptions.filter(
-				( newAttr ) =>
-					! variationOptions.find(
-						( attr: ProductAttribute ) =>
-							getAttributeId( newAttr ) === getAttributeId( attr )
-					)
-			),
-		] );
+		const addedAttributesOnly = newOptions.filter(
+			( newAttr ) =>
+				! variationOptions.some(
+					( attr: ProductAttribute ) =>
+						getAttributeId( newAttr ) === getAttributeId( attr )
+				)
+		);
+		recordEvent( 'product_options_add', {
+			source: TRACKS_SOURCE,
+			options: addedAttributesOnly.map( ( attribute ) => ( {
+				attribute: attribute.name,
+				values: attribute.options,
+			} ) ),
```

Comment:
Is it necessary to send all the options to tracks? This seems a bit overkill to me, and also kinda intrusive when it comes to data collection.
Couldn't we instead collect the amount of options -> `optionsLength` and also the amount of `values` might be selected per attribute?
Collecting the attribute name seems rather useless to me, the only one I could see of benefit is the `name` length, but I guess we would add that to the option create track.
cc: @AnnaMag

Reply:
> This seems a bit overkill to me, and also kinda intrusive when it comes to data collection.

The users explicitly opt-in and that data can be made "Visible to customers" by user's choice. The median number of terms is 1, so the data set is not large. It is a valid concern, but we are not using personal data.

> Collecting the attribute name seems rather useless to me 

I can see a few possible use cases, e.g. fine-tune an AI model for automated suggestions.

Reply:
> It is a valid concern, but we are not using personal data.

The personal data is more so the option names and the attribute name.
I also still not sure if the `opt-in` gives us permission to start collecting any and all data, given we also have this specific page listing what we track: https://woocommerce.com/usage-tracking/ (but I am not sure how up to date that page is).

> I can see a few possible use cases, e.g. fine-tune an AI model for automated suggestions or product creation.

AI generation is a good suggestion, although that makes me wonder if that would require a separate `opt-in` to allow data to be used for AI, or update our current opt-in to allow for that.